src/sidebar/components/search-status-bar.js

```diff
@@ -1,63 +1,121 @@
 'use strict';
 
-const memoize = require('../util/memoize');
+const { Fragment, createElement } = require('preact');
+const propTypes = require('prop-types');
+const { useMemo } = require('preact/hooks');
 
+const { withServices } = require('../util/service-context');
 const uiConstants = require('../ui-constants');
+const useStore = require('../store/use-store');
 
-// @ngInject
-function SearchStatusBarController(store, rootThread) {
-  this.TAB_ANNOTATIONS = uiConstants.TAB_ANNOTATIONS;
-  this.TAB_NOTES = uiConstants.TAB_NOTES;
-  this.TAB_ORPHANS = uiConstants.TAB_ORPHANS;
+const countVisibleAnns = annThread => {
+  return annThread.children.reduce(
+    function(count, child) {
+      return count + countVisibleAnns(child);
+    },
+    annThread.visible ? 1 : 0
+  );
+};
 
-  const thread = () => {
-    return rootThread.thread(store.getState());
-  };
+/**
+ * A bar where the user can clear a selection or search and see whether
+ * any search results were found.
+ * */
+function SearchStatusBar({
+  selectedTab,
+  totalAnnotations,
+  totalNotes,
+  rootThread,
+}) {
+  const storeState = useStore(store => store.getState());
+  const clearSelection = useStore(store => store.clearSelection);
+  const filterQuery = storeState.filterQuery;
+  const filterActive = !!storeState.filterQuery;
+
+  const thread = rootThread.thread(storeState);
 
-  const visibleCount = memoize(thread => {
-    return thread.children.reduce(
-      function(count, child) {
-        return count + visibleCount(child);
-      },
-      thread.visible ? 1 : 0
-    );
-  });
-
-  this.filterMatchCount = function() {
-    return visibleCount(thread());
+  const visibleCount = useMemo(() => {
+    return countVisibleAnns(thread);
+  }, [thread]);
+
+  const filterResults = () => {
+    const resultsCount = visibleCount;
+    switch (resultsCount) {
+      case 0:
+        return 'No results for "' + filterQuery + '"';
+      case 1:
+        return '1 search result';
+      default:
+        return resultsCount + ' search results';
+    }
   };
 
-  this.areAllAnnotationsVisible = function() {
-    if (store.getState().directLinkedGroupFetchFailed) {
+  const areAllAnnotationsVisible = () => {
```

Comment:
Now that I look again, I think the naming here is inverted. Conceptually, we want to show a button to show all annotations/notes only if *not all annotations are visible*, eg. due to there being an active selection. This issue applied to the Angular code.

Reply:
Hmmm...yeah you're right.